### PR TITLE
feat: [TKC-2882] stream service  and parallel step logs

### PR DESCRIPTION
**Pull request by vsukhin**
## Pull request description 

- add api endpoints to stream service and parallel step logs using websockets and sse
- add cli command parameters to watch service and parallel step logs
- add new client methods
- add grpc method to stream service and parallel step logs
- add proto methods

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [x] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: internal/app/api/v1/testworkflowexecutions.go [hunk 1]
```diff
@@ -23,6 +24,41 @@ import (
 	"github.com/kubeshop/testkube/pkg/testworkflows/executionworker/executionworkertypes"
 )
 
+func (s *TestkubeAPI) streamNotifications(ctx *fasthttp.RequestCtx, id string, notifications executionworkertypes.NotificationsWatcher) {
+	// Initiate processing event stream
+	ctx.SetContentType("text/event-stream")
```
[the remark this answers is not in the document]
Comment (vsukhin) [reply internal/app/api/v1/testworkflowexecutions.go:29]: We use SSE in CLI and Web Socket in UI

File: pkg/agent/testworkflows.go [hunk 2]
```diff
@@ -162,6 +240,78 @@ func (ag *Agent) executeWorkflowNotificationsRequest(ctx context.Context, req *c
...
 }
 
+func (ag *Agent) executeWorkflowServiceNotificationsRequest(ctx context.Context, req *cloud.TestWorkflowServiceNotificationsRequest) error {
+	var (
+		notificationsCh <-chan testkube.TestWorkflowExecutionNotification
+		err             error
+	)
+
+	for {
+		notificationsCh, err = ag.testWorkflowServiceNotificationsFunc(ctx, req.ExecutionId, req.ServiceName, int(req.ServiceIndex))
```
[the remark this answers is not in the document]
Comment (vsukhin) [reply pkg/agent/testworkflows.go:246]: Yes, sure. Just want to make sure it works in general )

File: cmd/api-server/main.go [hunk 3]
```diff
@@ -325,6 +325,48 @@ func main() {
 		}
 		return notifications.Channel(), nil
 	}
+	getTestWorkflowServiceNotificationsStream := func(ctx context.Context, executionID, serviceName string, serviceIndex int) (<-chan testkube.TestWorkflowExecutionNotification, error) {
```
Comment (exu) [inline cmd/api-server/main.go:328]: I would think about extracting this to separate package - to not make main too big.

**Review by exu — APPROVED** [review]

File: cmd/api-server/main.go (hunk 3 above)
Comment (rangoo94) [reply cmd/api-server/main.go:328]: side note: `main` will be split soon to smaller `main`s, so I guess it's not a big deal 🙂

Comment (vsukhin) [reply cmd/api-server/main.go]: adressed

**Pull request merged**